core/dbt/adapters/factory.py

```diff
@@ -214,6 +217,10 @@ def get_adapter_type_names(name: Optional[str]) -> List[str]:
     return FACTORY.get_adapter_type_names(name)
 
 
+def get_adapter_constraint_support(name: str) -> List[str]:
```

Comment:
nit: looks like `Optional[str]` would be more accurate given usage in https://github.com/dbt-labs/dbt-core/pull/7476/files#diff-42cfc99b2fcd89ab75957b8999bee2b1af822f7d39953ca99f1f285c170557c1R630